code/src/java/pcgen/cdom/facet/base/AbstractItemFacet.java

```diff
@@ -110,9 +110,23 @@ public T remove(IDT id)
 	 * @return the item value for this AbstractItemFacet and the Player
 	 *         Character represented by the given PCGenIdentifier.
 	 */
-	public T get(IDT id)
+	public T get(final IDT id)
+	{
+		final T tmp = (T)getCache(id);
```

Comment:
I can easily remove the final style change. What do you think of the architectural approach though?